core/embed/trezorhal/stm32u5/flash.c

```diff
@@ -29,7 +29,13 @@
 #define FLASH_SEC_START_ADDRESS 0x0C000000
 #define FLASH_START_ADDRESS 0x08000000
 
-#define FLASH_SECTOR_COUNT (256 * 2)
+#ifdef STM32U585xx
+#define FLASH_BANK_PAGES 128
```

Comment:
We coould utilize constants from the stm32u5xxxx.h headers and not introduce our own. The code would be more universal then. See FLASH_SIZE, FLASH_BANK_SIZE, FLASH_PAGE_SIZE and FLASH_PAGE_NB.

Reply:
Slightly against it, the register with flash size is somewhere behind the OTP area, which would require expanding this MPU region + adding it to boardloader.

Removed the custom FLASH_START constants, see https://github.com/trezor/trezor-firmware/commit/ecd4585ab655d0fa6cdcae6f764fa3e3739e553d

Reply:
Okay, I understand. So let the other constants be.